src/message/socket-wrapper.js

```diff
@@ -40,6 +42,53 @@ var SocketWrapper = function( socket, options ) {
 };
 
 utils.inherits( SocketWrapper, EventEmitter );
+SocketWrapper.lastPreparedMessage = null;
+
+/**
+ * Updates lastPreparedMessage and returns the [uws] prepared message.
+ *
+ * @public
+ * @returns {External} prepared message
+ */
+SocketWrapper.prepareMessage = function( message ) {
+	SocketWrapper.lastPreparedMessage = message;
+	return uws.native.server.prepareMessage( message, uws.OPCODE_TEXT );
+}
+
+/**
+ * Sends the [uws] prepared message, or in case of testing sends the
+ * last prepared message.
+ *
+ * @public
+ * @returns {void}
+ */
+SocketWrapper.prototype.sendPrepared = function(preparedMessage) {
+	if ( this.socket.external ) {
```

Comment:
So its a code branch specifically written for test scenarios. Does this only matter in the UTs for socket wrapper? everything else should use a mock socket wrapper anyways. Using test codepaths in code is generally confusing and not something we have anywhere in any of our repos.